cloud/services/disks/disks_test.go

```diff
@@ -140,15 +145,16 @@ func TestDeleteDisk(t *testing.T) {
 
 			clusterScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
 				AzureClients: scope.AzureClients{
-					SubscriptionID: "123",
+					SubscriptionID: subscriptionID,
```

Comment:
```suggestion
```
you shouldn't need to set `SubscriptionID`both in `AzureClients` and `AzureCluster.Spec`, only setting it in `AzureCluster.Spec` should be enough (and the one set on AzureClients is a no-op since we're not checking for it in setCredentials anymore). 

Same comment for other test files.